src/prommis/uky/uky_flowsheet.py

```diff
@@ -2362,146 +2519,43 @@ def add_costing(m):
         },
     )
 
-    # 9.5 is UKy Rare Earth Element Precipitation - Filter Press
     # 10.4 is UKy Oxalate Precipitation - Filter Press
-    reep_filter_press_accounts = ["9.5", "10.4"]
-    m.fs.reep_filter_press = UnitModelBlock()
-    m.fs.reep_filter_press.volume = Var(initialize=0.405, units=units.ft**3)
-    m.fs.reep_filter_press.volume.fix()
-    m.fs.reep_filter_press.costing = UnitModelCostingBlock(
+    reep_filter_press_accounts = ["10.4"]
+    m.fs.sl_sep2.volume = Var(initialize=0.405, units=units.ft**3, bounds=(0, None))
+
+    @m.fs.sl_sep2.Constraint(reep_filter_press_accounts)
+    def volume_scaling_constraint(c, k):
+        return m.fs.sl_sep2.volume == units.convert(
+            REE_costing_params["1"][k]["RP Value"]
+            * units.ft**3
+            * (
+                m.fs.precipitator.aqueous_inlet.flow_vol[0]
+                / reference_basis_flow["precipitator_solex_aqueous_flow_vol"]
+            ),
+            to_units=units.ft**3,
+        )
+
+    m.fs.sl_sep2.costing = UnitModelCostingBlock(
         flowsheet_costing_block=m.fs.costing,
         costing_method=QGESSCostingData.get_REE_costing,
         costing_method_arguments={
             "cost_accounts": reep_filter_press_accounts,
-            "scaled_param": m.fs.reep_filter_press.volume,
+            "scaled_param": m.fs.sl_sep2.volume,
             "source": 1,
             "n_equip": 1,
             "scale_down_parallel_equip": False,
             "CE_index_year": CE_index_year,
         },
     )
 
-    # 9.8 is UKy Rare Earth Element Precipitation - Roaster
     # 10.5 is UKy Oxalate Precipitation - Roaster
-    reep_roaster_accounts = ["9.8", "10.5"]
-    m.fs.reep_roaster = UnitModelBlock()
-    m.fs.reep_roaster.duty = Var(initialize=0.035, units=units.MBTU / units.hr)
-    m.fs.reep_roaster.duty.fix()
-    m.fs.reep_roaster.costing = UnitModelCostingBlock(
+    reep_roaster_accounts = ["10.5"]
+    m.fs.roaster.costing = UnitModelCostingBlock(
         flowsheet_costing_block=m.fs.costing,
         costing_method=QGESSCostingData.get_REE_costing,
         costing_method_arguments={
             "cost_accounts": reep_roaster_accounts,
-            "scaled_param": m.fs.reep_roaster.duty,
-            "source": 1,
-            "n_equip": 1,
-            "scale_down_parallel_equip": False,
-            "CE_index_year": CE_index_year,
-        },
-    )
-
-    # Roasting costs
-    # 3.1 is UKy Roasting - Storage Bins
-    R_storage_bins_accounts = ["3.1"]
-    m.fs.R_storage_bins = UnitModelBlock()
```

Comment:
Why are all the roasting costs being deleted? Are they all being accounted for in 10.5?

Reply:
These accounts are the pre-leaching roaster, which we don't have in this flowsheet. 10.5 is the post-oxalate precipitation roaster, and we kept that since that is included in this flowsheet.